pandas/io/formats/style.py

```diff
@@ -802,6 +810,146 @@ def where(
             lambda val: value if cond(val) else other, subset=subset, **kwargs
         )
 
+    def set_tooltips(self, ttips: DataFrame):
+        """
+        Add string based tooltips that will appear in the `Styler` HTML result.
+
+        Parameters
+        ----------
+        ttips : DataFrame
+            DataFrame containing strings that will be translated to tooltips. Empty
+            strings, None, or NaN values will be ignored. DataFrame must have
+            identical rows and columns to the underlying `Styler` data.
+
+        Returns
+        -------
+        self : Styler
+
+        Notes
+        -----
+        Tooltips are created by adding `<span class="pd-t"></span>` to each data cell
+        and then manipulating the table level CSS to attach pseudo hover and pseudo
+        after selectors to produce the required the results. For styling control
+        see `:meth:Styler.set_tooltips_class`.
+        Tooltips are not designed to be efficient, and can add large amounts of
+        additional HTML for larger tables, since they also require that `cell_ids`
+        is forced to `True`.
+        """
+        if not (self.columns.equals(ttips.columns) and self.index.equals(ttips.index)):
+            raise AttributeError(
+                "Tooltips DataFrame must have identical column and index labelling "
+                "to underlying."
+            )
+
+        self.cell_ids = True  # tooltips only work with individual cell_ids
```

Comment:
isn't this an error if self.cell_ids is not None  (already)?

Reply:
`cell_ids` is True by default. If the user specifically changes `cell_ids` to False, maybe it is better to raise NotImplementedError for tooltips since the current code doesn't perform optimisation by cell (although is possible).  (see the commit..)